homeassistant/components/reolink/host.py

```diff
@@ -0,0 +1,168 @@
+"""This component encapsulates the NVR/camera API and subscription."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+import aiohttp
+from reolink_ip.api import Host
+from reolink_ip.exceptions import (
+    ApiError,
+    CredentialsInvalidError,
+    InvalidContentTypeError,
+)
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_PROTOCOL, CONF_USE_HTTPS, DEFAULT_PROTOCOL, DEFAULT_TIMEOUT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ReolinkHost:
+    """The implementation of the Reolink Host class."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: dict,
+        options: dict,
+    ) -> None:
+        """Initialize Reolink Host. Could be either NVR, or Camera."""
+        self._hass: HomeAssistant = hass
+
+        self._clientsession: aiohttp.ClientSession | None = None
+        self._unique_id: str | None = None
```

Comment:
I'd recommend refactoring so that the `unique_id` attribute is not optional, to make this class more convenient. We know that there will be a unique id if we succeed to initialize the api.